xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,1052 @@
+"""Utils for balance_changes and orderbook_changes."""
+from __future__ import annotations
+
+from datetime import datetime
+from decimal import Decimal
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+
+from pydash import compact, filter_, flatten, group_by, map_, map_values  # type: ignore
+from typing_extensions import Literal
+
+from xrpl.constants import XRPLException
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+"""
+###########################################
+Utils balance_parser and orderbook_parser
+###########################################
+"""
+
+
+# metadata received from a tx method
+METADATA_TYPE = Dict[
+    str,
+    Union[
+        str,
+        int,
+        Dict[str, str],  # issued currency amount
+        List[List[Dict[str, Union[str, int]]]],  # Field: 'paths'
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+# metadata received from a subscribtion method
+SUBSCRIPTION_METADATA_TYPE = Dict[
+    str,
+    Union[
+        str,
+        int,
+        Dict[  # Field: 'transaction'
+            str,
+            Union[
+                str,
+                int,
+                Dict[str, str],  # issued currency amount
+                List[List[Dict[str, Union[str, int]]]],  # Field: 'paths'
+            ],
+        ],
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+class XRPLMetadataException(XRPLException):
+    """Exception for invalid transaction metadata."""
+
+    pass
+
+
+def is_valid_metadata(
```

Comment:
```suggestion
def validate_metadata(
```
Using the `is_valid` implies that it's going to be returning a boolean as to whether the metadata is valid or not, but that's not what this method does.